code/global.dm

```diff
@@ -119,4 +119,7 @@ var/max_explosion_range = 14
 // Announcer intercom, because too much stuff creates an intercom for one message then hard del()s it.
 var/global/obj/item/device/radio/intercom/global_announcer = new(null)
 
+var/global/gun_serial_id = 485012 // incremented for each new gun . used for unique serials
+								// starts at 485012 cause ancap i guess
```

Comment:
Maybe need a separate serial for each gun type or at least dependent on manufacturer.

Reply:
why ? its a pain to code that for a simple feature.

Reply:
Because every gun on the ship having nearly identical (yet never matching) serial numbers, regardless of manufacturer, makes no sense.

Reply:
give me a random number generating algorithm then